specifyweb/frontend/js_src/lib/components/WorkBench/hooks.ts

```diff
@@ -30,56 +33,66 @@ export function getHotHooks(wbView: WbView) {
      * cases
      *
      */
-    afterRenderer: (td, visualRow, visualCol, property, _value) => {
-      if (wbView.hot === undefined) {
+    afterRenderer: (
+      td: any,
+      visualRow: number,
+      visualCol: number,
+      property: number | string,
+      _value: string
+    ) => {
+      if (workbench.hot === undefined) {
         td.classList.add('text-gray-500');
         return;
       }
-      const physicalRow = wbView.hot.toPhysicalRow(visualRow);
+      const physicalRow = workbench.hot.toPhysicalRow(visualRow);
       const physicalCol =
         typeof property === 'number'
           ? property
-          : wbView.hot.toPhysicalColumn(visualCol);
-      if (physicalCol >= wbView.dataset.columns.length) return;
-      const metaArray = wbView.cells.cellMeta?.[physicalRow]?.[physicalCol];
-      if (wbView.cells.getCellMetaFromArray(metaArray, 'isModified'))
-        wbView.cells.runMetaUpdateEffects(
+          : workbench.hot.toPhysicalColumn(visualCol);
+      if (physicalCol >= workbench.dataset.columns.length) return;
+      const metaArray = workbench.cells?.cellMeta?.[physicalRow]?.[physicalCol];
+      if (workbench.cells?.getCellMetaFromArray(metaArray!, 'isModified'))
+        workbench.cells.runMetaUpdateEffects(
           td,
           'isModified',
           true,
           visualRow,
           visualCol
         );
-      if (wbView.cells.getCellMetaFromArray(metaArray, 'isNew'))
-        wbView.cells.runMetaUpdateEffects(
+      if (workbench.cells?.getCellMetaFromArray(metaArray!, 'isNew'))
+        workbench.cells?.runMetaUpdateEffects(
           td,
           'isNew',
           true,
           visualRow,
           visualCol
         );
-      if (wbView.cells.getCellMetaFromArray(metaArray, 'isSearchResult'))
-        wbView.cells.runMetaUpdateEffects(
+      if (workbench.cells!.getCellMetaFromArray(metaArray!, 'isSearchResult'))
+        workbench.cells?.runMetaUpdateEffects(
```

Comment:
if you have `workbench.cells!` in the condition, then I think typescript does not require `?` in the condition body as you already told it that `I know better than you, this is not null`

```suggestion
        workbench.cells.runMetaUpdateEffects(
```

though, you used `?` instead of `!` on line 62 and 54 instead - need to be consistent)

also, could you instead have a single `if (workbench.cells === undefined) return;` check at the top of `afterRenderer` so that you don't have to do that that many times inside?
(plus, each needless `?` is not free in terms of performance, and this function is in the hot path as it's called for every single rendered cell in the spreadsheet)